discoverer/src/main/java/eu/seaclouds/platform/discoverer/core/Discoverer.java

```diff
@@ -17,39 +17,51 @@
 
 package eu.seaclouds.platform.discoverer.core;
 
-import eu.seaclouds.platform.discoverer.api.*;
+import com.mongodb.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import eu.seaclouds.platform.discoverer.crawler.CloudHarmonySPECint;
 import eu.seaclouds.platform.discoverer.crawler.CrawlerManager;
-import io.dropwizard.Application;
-import io.dropwizard.setup.Environment;
+import org.bson.Document;
 
+import java.io.*;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Date;
 
 
-public class Discoverer extends Application<DiscovererConfiguration> {
+public class Discoverer {
     /* singleton */
-    private static final Discoverer singleInstance = new Discoverer();
     private ArrayList<String> offeringNodeTemplates = new ArrayList<>();
     private boolean refreshing = false;
     private ArrayList<String> activeCrawlers;
 
+    private String databaseName = "DrACO_DB";
+    private String collectionName = "offerings";
+
     public int totalCrawledOfferings;
     public int crawledTimes;
     public Date lastCrawl;
 
-    public static Discoverer instance() { return singleInstance; }
+    public Discoverer(MongoClient mongoClient, ArrayList<String> activeCrawlers) {
+        this.initializeResources();
 
-    /* vars */
-    public OfferingManager offeringManager;
+        MongoDatabase db = mongoClient.getDatabase(this.databaseName);
+        MongoCollection<Document> coll = db.getCollection(this.collectionName);
+
+        this.offeringManager = new OfferingManager(coll);
+        this.offeringManager.initializeOfferings();
 
-    /* *************************************************************** */
-    /* **                     subparts initialization               ** */
-    /* *************************************************************** */
-    public Discoverer() {
-        this.offeringManager = new OfferingManager();
+        this.activeCrawlers = activeCrawlers;
     }
 
+    public Discoverer(MongoClient mongoClient) {
+        this(mongoClient, new ArrayList<String>());
+    }
+
+    /* vars */
+    public OfferingManager offeringManager;
```

Comment:
What is the role of this `OfferingManager`? is it a [DAO](https://en.wikipedia.org/wiki/Data_access_object) maybe?

If so, mabe something similar to what it is suggested on Dropwizard website ( [JDBI](http://www.dropwizard.io/0.9.2/docs/manual/jdbi.html) or [Hibernate](http://www.dropwizard.io/0.9.2/docs/manual/hibernate.html)) may be what you want to use to redesign the Discoverer.

Thoughts?